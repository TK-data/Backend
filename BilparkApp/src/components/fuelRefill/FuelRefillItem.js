import React from 'react';
import { StyleSheet, View, Text, Alert } from 'react-native';
import { Button } from 'native-base';
import { connect } from 'react-redux';
import { deleteFuelRefill } from '../../actions/fuelRefill';


const styles = StyleSheet.create({
  text: {
    color: 'white',
    alignSelf: 'flex-start',
  },
  texts: {
    flexDirection: 'column',
    alignSelf: 'center',
  },
  infoText: {
    color: 'white',
    fontSize: 20,
    alignSelf: 'center',
  },
  priceText: {

  },
  content: {
    flex: 1,
    justifyContent: 'space-between',
    flexDirection: 'row',
    margin: 15,
  },
  button: {
    paddingLeft: 10,
    paddingRight: 10,
  },
});

const weekDays = {
  0: 'Søndag',
  1: 'Mandag',
  2: 'Tirsdag',
  3: 'Onsdag',
  4: 'Torsdag',
  5: 'Fredag',
  6: 'Lørdag',
};

const months = {
  0: 'januar',
  1: 'februar',
  2: 'mars',
  3: 'april',
  4: 'mai',
  5: 'juni',
  6: 'juli',
  7: 'august',
  8: 'september',
  9: 'oktober',
  10: 'november',
  11: 'desember',
};


const FuelRefillItem = ({ FuelTime, RefillID, deleteItem, Rate, Price }) => {

  const fuelDate = new Date(FuelTime);

  const month = months[fuelDate.getMonth()];
  const date = fuelDate.getDate();
  const day = weekDays[fuelDate.getDay()];
  const hours = fuelDate.getHours() > 9 ? fuelDate.getHours() : '0' + fuelDate.getHours();
  const minutes = fuelDate.getMinutes() > 9 ? fuelDate.getMinutes() : '0' + fuelDate.getMinutes();

  return (
    <View style={styles.content}>
      <View style={styles.texts}>
        <Text style={styles.text}>
          {day} {date}. {month}
        </Text>
        <Text style={styles.text}>
          kl. {hours}:{minutes}
        </Text>
      </View>
      <Text style={styles.infoText}>
        {Price} kr @ {Rate} kr/l
      </Text>
      <Button
        light
        style={styles.button}
        onPress={() => {
          Alert.alert(
            'Bekreft sletting',
            'Er du sikker på at du vil slette loggføringen?',
            [
              { text: 'Avbryt', style: 'cancel' },
              { text: 'Slett', onPress: () => deleteItem(RefillID) },
            ],
          );
        }}
      >
        <Text>
          slett
        </Text>
      </Button>
    </View>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    deleteItem: (RefillID) => {
      dispatch(deleteFuelRefill(RefillID));
    },
  };
};

const mapStateToProps = state => ({
});

export default connect(mapStateToProps, mapDispatchToProps)(FuelRefillItem);
