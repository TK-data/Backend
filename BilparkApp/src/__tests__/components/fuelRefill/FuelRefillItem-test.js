import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import FuelRefillItem from '../../../components/fuelRefill/FuelRefillItem';

const initialState = {
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing FuelRefillItem', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <FuelRefillItem />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
