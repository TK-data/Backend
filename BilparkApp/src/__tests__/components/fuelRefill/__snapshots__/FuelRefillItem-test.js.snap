// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FuelRefillItem Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FuelRefillItem
    deleteItem={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "column",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "white",
              }
            }
          >
             
            NaN
            . 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "white",
              }
            }
          >
            kl. 
            0NaN
            :
            0NaN
          </Text>
        </View>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 20,
            }
          }
        >
           kr @ 
           kr/l
        </Text>,
        <Styled(Button)
          light={true}
          onPress={[Function]}
          style={
            Object {
              "paddingLeft": 10,
              "paddingRight": 10,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            slett
          </Text>
        </Styled(Button)>,
      ],
      "style": Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 15,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-start",
                  "color": "white",
                }
              }
            >
               
              NaN
              . 
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-start",
                  "color": "white",
                }
              }
            >
              kl. 
              0NaN
              :
              0NaN
            </Text>,
          ],
          "style": Object {
            "alignSelf": "center",
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                undefined,
                " ",
                NaN,
                ". ",
                undefined,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "flex-start",
                "color": "white",
              },
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " ",
              NaN,
              ". ",
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "kl. ",
                "0NaN",
                ":",
                "0NaN",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "flex-start",
                "color": "white",
              },
            },
            "ref": null,
            "rendered": Array [
              "kl. ",
              "0NaN",
              ":",
              "0NaN",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            undefined,
            " kr @ ",
            undefined,
            " kr/l",
          ],
          "ellipsizeMode": "tail",
          "style": Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 20,
          },
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " kr @ ",
          undefined,
          " kr/l",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            slett
          </Text>,
          "light": true,
          "onPress": [Function],
          "style": Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "slett",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "slett",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignSelf": "center",
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-start",
                  "color": "white",
                }
              }
            >
               
              NaN
              . 
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-start",
                  "color": "white",
                }
              }
            >
              kl. 
              0NaN
              :
              0NaN
            </Text>
          </View>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 20,
              }
            }
          >
             kr @ 
             kr/l
          </Text>,
          <Styled(Button)
            light={true}
            onPress={[Function]}
            style={
              Object {
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              slett
            </Text>
          </Styled(Button)>,
        ],
        "style": Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 15,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "color": "white",
                  }
                }
              >
                 
                NaN
                . 
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "color": "white",
                  }
                }
              >
                kl. 
                0NaN
                :
                0NaN
              </Text>,
            ],
            "style": Object {
              "alignSelf": "center",
              "flexDirection": "column",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  undefined,
                  " ",
                  NaN,
                  ". ",
                  undefined,
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "flex-start",
                  "color": "white",
                },
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " ",
                NaN,
                ". ",
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "kl. ",
                  "0NaN",
                  ":",
                  "0NaN",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "flex-start",
                  "color": "white",
                },
              },
              "ref": null,
              "rendered": Array [
                "kl. ",
                "0NaN",
                ":",
                "0NaN",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              undefined,
              " kr @ ",
              undefined,
              " kr/l",
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 20,
            },
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " kr @ ",
            undefined,
            " kr/l",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              slett
            </Text>,
            "light": true,
            "onPress": [Function],
            "style": Object {
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "slett",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "slett",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
